src/CanBeSliced.php

```diff
@@ -14,21 +14,17 @@
 
     abstract public function slice(int $offset, int $length = null) : Sequence;
 
-    final private function getPage(int $page) : array
+    final private function getPage(int $page)
     {
-        if (empty($this->pages)) {
-            for ($i = 0; $i < $this->count(); ++$i) {
-                if (0 === $i % $this->pageBatch) {
-                    $this->pages[count($this->pages) + 1] = $this->slice($i, $this->pageBatch);
-                }
-            }
+        if (empty($this->pages[$page])) {
+            $this->pages[$page] = $this->slice($this->pageBatch * ($page - 1), $this->pageBatch);
         }
 
         if (false === isset($this->pages[$page])) {
             throw new LogicException('Could not find page '.$page);
         }
 
-        return iterator_to_array($this->pages[$page]);
+        return $this->pages[$page]->toArray();
```

Comment:
so slice() is returning a type-hinted `Sequence` which extends `Collection` which has `toArray()`. The only call is in `SlicedIterator`, and a `Sequence` is an `ArrayAccess` anyway; I'd say we can return a `Sequence`.